src/store/ethereum/index.ts

```diff
@@ -183,6 +186,15 @@ async function setWalletType(context: ActionContext, walletType: string) {
   }
 }
 
+function walletDisconnectBeforeUnload(wallet) {
+  const listener = (e: BeforeUnloadEvent) => {
+    e.preventDefault()
+    wallet.disconnect()
+    window.removeEventListener("beforeunload", listener)
+  }
+  window.addEventListener("beforeunload", listener)
+}
```

Comment:
So this is no longer necessary then?

Reply:
Does not hurt to keep it for other wallets